### PR TITLE
 support state level instructional resources for user created groups

### PR DESCRIPTION
**Pull request by dsnyder-fairway**

### CODE REVIEW

File: reporting-service/src/main/resources/application-no-permission.sql.yml [hunk 1]
```diff
@@ -200,6 +200,34 @@ sql:
...
+          ON ir.org_level = 'DistrictGroup'
+          AND district_group.id = ir.org_id
+        LEFT JOIN district
+          ON ir.org_level = 'District'
+          AND district.id = ir.org_id
+        LEFT JOIN school_group
+          ON ir.org_level = 'SchoolGroup'
+          AND school_group.id = ir.org_id
+      WHERE asmt.id = :asmt_id
+        AND
```
Comment (jtreuting) [inline reporting-service/src/main/resources/application-no-permission.sql.yml]: I don't think we need a second SQL query and could use the main one and pass in school_id NULL.  Should be able to edit the line `AND school.id = :school_id` to be `AND (:school_id IS NULL OR school.id = :school_id)`

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/JdbcInstructionalResourceRepository.java [hunk 2]
```diff
@@ -41,4 +44,20 @@ public JdbcInstructionalResourceRepository(final NamedParameterJdbcTemplate temp
                                 .build()
         );
     }
+
+    @Override
+    public Collection<InstructionalResource> findByAssessment(final long assessmentId) {
```
Comment (jtreuting) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/JdbcInstructionalResourceRepository.java]: Also think we could have a single method with schoolId being NULL

File: reporting-service/src/main/resources/application-no-permission.sql.yml (hunk 1 above)
Comment (dsnyder-fairway) [reply reporting-service/src/main/resources/application-no-permission.sql.yml]: that was my first attempt, but that resulted in 8 rows being returned for a single instructional resource

Comment (dsnyder-fairway) [reply reporting-service/src/main/resources/application-no-permission.sql.yml]: I had to `UNION` these two selects. See https://stackoverflow.com/a/1255523

**Comment by dsnyder-fairway** [comment]
Attn @agorina

**Pull request merged**